crates/common_utils/src/events.rs

```diff
@@ -125,6 +128,8 @@ impl_api_event_type!(
         String,
         id_type::MerchantId,
         (id_type::MerchantId, String),
+        (id_type::MerchantId, id_type::ApiKeyId),
+        (&id_type::MerchantId, &id_type::ApiKeyId),
```

Comment:
In these cases, we can raise `ApiEventsType::ApiKey` itself, these implementations can go in the `crates/common_utils/src/id_type/api_key.rs` file for now.